### PR TITLE
[Mempool] CheckSigOps using incorrect constant (#2470)

### LINKED ISSUE #2470: [Mempool] CheckSigOps using incorrect constant
Used in https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs#L768 ->

`if (context.SigOpsCost > this.ConsensusOptions.MaxBlockSigopsCost)`

Used in core https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L724 -> 

`if (nSigOpsCost > MAX_STANDARD_TX_SIGOPS_COST)`

These are two distinct constants with their own values (the max for blocks is 80 000 and the max for a single transaction is 80 000 / 5 = 16 000).

### PR DESCRIPTION
**Pull request by MithrilMan**
Added maxStandardTxSigopsCost ConsensusOptions property to fix #2470

### CODE REVIEW

**Comment by bokobza** [comment]
In fact, hold up.
@MithrilMan Are you sure MaxBlockSigopsCost is not used instead of MaxStandardTxSigopsCost in other part of the code?

**Comment by MithrilMan** [comment]
check on bitcoin code
https://github.com/bitcoin/bitcoin/search?utf8=%E2%9C%93&q=MAX_STANDARD_TX_SIGOPS_COST&type=

evidences that it's used only there. Not sure if this is a valid proof

**Comment by bokobza** [comment]
Look at the stratisX code as well.

File: src/NBitcoin/ConsensusOptions.cs [hunk 1]
```diff
@@ -52,6 +55,7 @@ public ConsensusOptions()
             this.MaxStandardTxWeight = 400000;
             this.MaxBlockBaseSize = 1000000;
             this.MaxBlockSigopsCost = 80000;
+            this.MaxStandardTxSigopsCost = this.MaxBlockSigopsCost / 5;
```
Comment (dangershony) [inline src/NBitcoin/ConsensusOptions.cs]: This is sort of a default value?

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs [hunk 2]
```diff
@@ -765,7 +765,7 @@ private void CheckSigOps(MempoolValidationContext context)
             // itself can contain sigops MAX_STANDARD_TX_SIGOPS is less than
             // MAX_BLOCK_SIGOPS; we still consider this an invalid rather than
             // merely non-standard transaction.
-            if (context.SigOpsCost > this.ConsensusOptions.MaxBlockSigopsCost)
+            if (context.SigOpsCost > this.ConsensusOptions.MaxStandardTxSigopsCost)
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs]: @zeptin found this bug so lets let him have a look as well, I think also @majikandy had issues here.

**Comment by dangershony** [comment]
>check on bitcoin code
Yes ot be sure lets check stratisX as well

File: src/NBitcoin/ConsensusOptions.cs (hunk 1 above)
Comment (MithrilMan) [reply src/NBitcoin/ConsensusOptions.cs]: on bitcoin yes, it is
https://github.com/bitcoin/bitcoin/blob/78dae8caccd82cfbfd76557f1fb7d7557c7b5edb/src/policy/policy.h#L30

Comment (MithrilMan) [reply src/NBitcoin/ConsensusOptions.cs]: `static const unsigned int MAX_STANDARD_TX_SIGOPS_COST = MAX_BLOCK_SIGOPS_COST/5;`

Comment (bokobza) [reply src/NBitcoin/ConsensusOptions.cs]: same on StratisX @dangershony

**Comment by majikandy** [comment]
yeah so your /5 makes it similar to the bitcoin code but that won't solve the bug that @sondreb and I both experienced (separately).  Where after staking on a new network you have lots of inputs to a largish value transactions (eg staking for a 1000 blocks) and then sending 1mil coins,  it has loads of inputs that when you debug show as ancestors....

yet on that mempool check - the sigops is 2...   but at mining time, the sig ops is 86,000 ish, way over the 20,000 stratis limit and so can't be mined....

futhermore it screws up the wallet a bit as it remains as unconfirmed whilst stuck in the memool, and deleting the mempool and resyncing (via api) doesn't even solve that

**Comment by MithrilMan** [comment]
I'd consider this separated issue, being this PoS and not PoW so bitcoin related
We could leave issue open to another PR

**Comment by majikandy** [comment]
Yes I agree it is a different issue - but it is covered on the issue this is linked to so either create a new more specific one from that or keep that one open as covering both the bit this fixes and awaiting another PR

**Comment by majikandy** [comment]
PoS might not be related.....  (or perhaps it is, good point)

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs (hunk 2 above)
Comment (zeptin) [reply src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs]: I'm pretty sure this `if` is now correct, at least as far as stratisX is concerned. I agree that Andy's issue is probably not resolved by this PR and still needs to be looked at.

**Pull request merged**